OpenRA.Game/Game.cs

```diff
@@ -172,7 +172,32 @@ internal static void StartGame(string mapUID, WorldType type)
 			GC.Collect();
 		}
 
-		public static bool IsHost
+		public static void RestartGame()
+		{
+			string uid = OrderManager.World.Map.Uid;
+			bool record = false;
+			string difficulty = OrderManager.LobbyInfo.GlobalSettings.Difficulty;
+			string gameSpeed = OrderManager.LobbyInfo.GlobalSettings.GameSpeedType;
+			if (OrderManager.Connection is ReplayRecorderConnection)
```

Comment:
you can simplify this down to `var record = OrderManager.Connection is ReplayRecorderConnection`.  Please also move it to just before the `JoinServer` call where it is used.